esm/src/api/controllers/demo.controller.js

```diff
@@ -0,0 +1,201 @@
+      )
+    );
+  } catch (err) {
+    res.status(500).json({
+      error: `${err}`,
+    });
+    console.log(chalk.redBright(`\nError saving ${item}: ${err}\n`));
+  }
+};
+
+/*=============================
+  operations for /demo/demoId
+==============================*/
+
+const getOneDemoItemController = async (req, res, next) => {
+  try {
+    let doc = await getOneDemoItemService(req.params.demoId);
+    if (doc) {
+      res.status(200).json({
+        _id: doc._id,
+        name: doc.name,
+        age: doc.age,
+        request: {
+          type: "GET",
+          description: `Url link to all ${item}s`,
+          url: `http://localhost:3000/${routeName}/`,
+        },
+      });
+      console.log(
+        chalk.greenBright(
+          `\nGET request successful! \n\n${item} url: http://localhost:3000/${routeName}/${doc._id}\n`
+        )
+      );
+    } else {
+      console.log(chalk.redBright("\nNo record found for provided ID\n"));
+      return res.status(404).json({
+        message: "No record found for provided ID",
+      });
+    }
+  } catch (err) {
+    res.status(500).json({
+      message: "Invalid ID",
+      error: `${err}`,
+    });
+    console.log(chalk.redBright(`\nError retriving ${item}: ${err}\n`));
+  }
+};
+
+const deleteDemoItemController = async (req, res, next) => {
+  try {
+    let doc = await deleteDemoItemService(req.params.demoId);
+    console.log(chalk.greenBright(`\n${item} DELETED successfully!\n`));
+    res.status(200).json({
+      message: `${item} deleted successfully!`,
+      request: {
+        type: "POST",
+        description: "Url link to make post request to",
+        url: `http://localhost:3000/${item}/`,
+        body: {
+          name: "String",
+          age: "Number",
+        },
+      },
+    });
+  } catch (err) {
+    res.status(500).json({
+      message: `Error deleting ${item}`,
+      error: `${err}`,
+    });
+    console.log(chalk.redBright(`\nError deleting ${item}: ${err}\n`));
+  }
+};
+
+const partialUpdateDemoItemController = async (req, res, next) => {
+  try {
+    let doc = await partialUpdateDemoItemService(req.params.demoId, req.body);
+    console.log(
+      chalk.greenBright(
+        `\nPATCH request for ID ${doc._id} successful! \n\nUpdated ${item} url: http://localhost:3000/${routeName}/${doc._id}\n`
+      )
+    );
+    return res.status(200).json({
+      message: "Patch request successful!",
+      request: {
+        type: "GET",
+        description: `Url link to updated ${item}`,
+        url: `http://localhost:3000/${routeName}/${doc._id}`,
+      },
+    });
+  } catch (error) {
+    res.status(500).json({
+      message: `Error updating ${item} property & value`,
+      error: `${err}`,
+    });
+    console.log(
+      chalk.redBright(`\nError updating ${item} property & value: ${err}\n`)
+    );
+  }
+};
+
+
+const fullUpdateDemoItemController = async (req, res) => {
```

Comment:
### Review comment 2
#
Do the same thing I suggested for patch for put too @samuko-things

But this time the name should be `updateDemoItemPropertyValuesController`

Emphasis on "update" and plural of "PropertyValues" in the new name.

(also remember to do this for services, routes and any other place you used "fullUpdate")

Reply:
### Review comment 3
#

Make PATCH request code come after PUT (everywhere)